pkg/scalers/gitlab_runner_scaler_test.go

```diff
@@ -0,0 +1,468 @@
+		},
+		{
+			name: "Error retrieving queue length",
+
+			pipelinePendingQueueLength:            0,
+			pipelineWaitingForResourceQueueLength: 0,
+			pipelineRunningQueueLength:            0,
+
+			targetPipelineQueueLength: 5,
+			expectedMetricValue:       0,
+			expectedActive:            false,
+			expectError:               true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Setup mock server
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if tc.expectError {
+					w.WriteHeader(http.StatusInternalServerError)
+					return
+				}
+
+				w.WriteHeader(http.StatusOK)
+
+				page := r.URL.Query().Get("page")
+
+				pipelines := make([]map[string]interface{}, 0)
+
+				// make only the first page return the length of pipelines, the rest should be empty for pagination to stop
+				if page != "1" {
+					_ = json.NewEncoder(w).Encode(pipelines)
+					return
+				}
+
+				// populate pipelines based on the status
+				status := r.URL.Query().Get("status")
+				switch status {
+				case string(pipelinePendingStatus):
+					for i := int64(0); i < tc.pipelinePendingQueueLength; i++ {
+						pipelines = append(pipelines, map[string]interface{}{
+							"id": i + 1,
+						})
+					}
+				case string(pipelineWaitingForResourceStatus):
+					for i := int64(0); i < tc.pipelineWaitingForResourceQueueLength; i++ {
+						pipelines = append(pipelines, map[string]interface{}{
+							"id": i + 1,
+						})
+					}
+				case string(pipelineRunningStatus):
+					for i := int64(0); i < tc.pipelineRunningQueueLength; i++ {
+						pipelines = append(pipelines, map[string]interface{}{
+							"id": i + 1,
+						})
+					}
+				}
+
+				_ = json.NewEncoder(w).Encode(pipelines)
+			}))
+			defer server.Close()
+
+			meta := &gitlabRunnerMetadata{
+				GitLabAPIURL:              mustParseURL(server.URL),
+				PersonalAccessToken:       "test-token",
+				TargetPipelineQueueLength: tc.targetPipelineQueueLength,
+				ProjectID:                 "12345",
+			}
+
+			scaler := gitlabRunnerScaler{
+				metadata:   meta,
+				httpClient: http.DefaultClient,
+				logger:     logr.Discard(),
+			}
+
+			metrics, active, err := scaler.GetMetricsAndActivity(context.Background(), "gitlab-runner-queue-length")
+			if tc.expectError {
+				assert.Error(t, err)
+				assert.Empty(t, metrics, "Expected no metrics")
+				assert.False(t, active, "Expected not active")
+			} else {
+				assert.NoError(t, err)
+				assert.Len(t, metrics, 1, "Expected one metric")
+				assert.Equal(t, float64(tc.expectedMetricValue), metrics[0].Value.AsApproximateFloat64(), "Expected metric value")
+				assert.Equal(t, tc.expectedActive, active, "Expected active")
+			}
+		})
+	}
+}
+
+func TestGitLabRunnerScaler_GetMetricSpecForScaling(t *testing.T) {
+	meta := &gitlabRunnerMetadata{
+		ProjectID:                 "12345",
+		TargetPipelineQueueLength: 5,
+		TriggerIndex:              0,
```

Comment:
It's something quite small but, could we use `TriggerIndex: 1` ? 0 is the default value so just using 1 we can also check that the `TriggerIndex` is correctly reflected in the metric name